fedora_elections/elections.py

```diff
@@ -144,11 +141,19 @@ def vote_range(election):
         max_range=max_selection)
 
     if form.validate_on_submit():
-
         if form.action.data == 'submit':
             for candidate in form:
                 if candidate.short_name in ['csrf_token', 'action']:
                     continue
+                if revote:
+                    vote = update(models.Vote).\
+                    where(models.Vote.candidate_id == candidate.short_name
+                    and models.Vote.voter == flask.g.fas_user.username
+                    and models.Vote.election == election.id).\
+                    values(value = candidate.data)
+                    SESSION.execute(vote)
+                    #break out of candidate loop
+                    continue
```

Comment:
This is wrong, what you are doing here is an SQL query which means you are not taking advantages of the ORM.
What you want is not to update the DB via SQL but instead to update the object Vote corresponding to this vote.